ai_genomics/pipeline/patent_data/add_full_cpc_ipc_codes.py

```diff
@@ -0,0 +1,86 @@
+"""
+This script adds the full list of CPC and IPC codes per patent number to the AI and Genomics patents dataset saved in S3.
+"""
+from ai_genomics.getters.data_getters import load_s3_data, save_to_s3
+from ai_genomics import logger, bucket_name as BUCKET_NAME
+import pandas as pd
+from typing import List
+
+from ai_genomics.utils.patent_data.get_ai_genomics_patents_utils import est_conn
+from ai_genomics.pipeline.patent_data.get_ai_genomics_patents import (
+    covert_list_of_codes_to_string,
+)
+
+AI_GENOMICS_PATENTS = load_s3_data(
+    BUCKET_NAME,
+    "inputs/patent_data/processed_patent_data/ai_genomics_patents_cpc_ipc_codes.csv",
+)
+
+
+def get_full_cpc_ipc_codes_query(
+    ai_genomics_publication_numbers: List[
+        str
+    ] = AI_GENOMICS_PATENTS.publication_number.to_list(),
+) -> str:
+    """Generates query to retrieve full list of CPC and IPC codes per patent publication number.
+    Args:
+        ai_genomics_publication_numbers (list): a list of AI and Genomics publication numbers
+    Returns:
+        full_cpc_ipc_q (str): The BigQuery query to retrive a full list of CPC and IPC codes
+    """
+    ai_genomics_publication_numbers_formatted = covert_list_of_codes_to_string(
+        ai_genomics_publication_numbers
+    )
+
+    full_cpc_ipc_q = (
+        "select publication_number, cpc.code as cpc_code, ipc.code as ipc_code "
+        "from patents-public-data.patents.publications, "
+        "UNNEST(cpc) cpc, UNNEST(ipc) ipc "
+        f"WHERE publication_number in ({ai_genomics_publication_numbers_formatted})"
+    )
+
+    return full_cpc_ipc_q
+
+
+def add_full_cpc_ipc_codes(
+    full_cpc_ipc_df: pd.DataFrame, ai_genomics_df: pd.DataFrame = AI_GENOMICS_PATENTS
+) -> pd.DataFrame:
+    """Adds full list of CPC and IPC codes per patent publication number to ai genomics dataset.
+    Args:
+        full_cpc_ipc_df (pd.DataFrame): Dataframe of ai genomics patent numbers, cpc and ipc codes
+        ai_genomics_df (pd.DataFrame): Dataframe of ai genomics patents
+    Returns:
+        ai_genomics_df_full_codes (pd.DataFrame): DataFrame of ai genomics patents with full list of cpc and ipc codes
```

Comment:
When you return a variable without a name (including an existing variable with some operation applied to it), then you only need to put the description:

```
Returns:
    DataFrame of ai genomics patents with full list of cpc and ipc codes
```